lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp

```diff
@@ -265,6 +266,10 @@ class TSNodeLowering : public NodeLowering {
       return LowerLeakyReluBackward(ir::NodeCast<ir::ops::LeakyReluBackward>(
           node, ir::OpKind(at::aten::leaky_relu_backward)));
     }
+    if (node->op().op == at::aten::log_softmax) {
```

Comment:
Wondering under what situation this file needs to be modified given not all operators have their lowering here. @asuhan

Reply:
@alanwaketan Operators which are "pure" simply work by falling through here: https://github.com/pytorch/pytorch/blob/771a5ae8af25bf6b2689e55bc1fab4bcdccb7732/lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp#L309

By "pure" I mean they only have operands and don't have additional parameters. For example, `log_softmax` has an axis, convolutions have strides, dilation etc.

An accurate view of what's lowered is here: https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/lazy_tensor_core/ts_native_functions.yaml. In `TSNodeLowering` we had already selected the intended subset of operators.

Note that if something falls through unintentionally (we have a bug in the higher layers), we'll most likely get an error from TorchScript because the call we emit will lack those additional parameters, assuming it even passes shape inference. It's a fall-through, but it's pretty safe.

Reply:
Thanks for the explanations. They all make perfect sense.